### PR TITLE
fix: add some error log when get config before load

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
Signed-off-by: Lloyd-Pottiger <yan1579196623@gmail.com>

### What problem does this PR solve?

Issue Number: ref #5585 

Problem Summary:

### What is changed and how it works?

Because the default of profiles config is set during compiling, there are no any errors when we use the config before we load the config from use config file. And it is a huge task to check each config whether it can be configured, so add some error logs.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Interpreters/Context.cpp [hunk 1]
```diff
@@ -1006,6 +1006,8 @@ ASTPtr Context::getCreateDatabaseQuery(const String & database_name) const
 
 Settings Context::getSettings() const
 {
+    if (!is_config_loaded)
```
Comment (ywqzzy) [inline dbms/src/Interpreters/Context.cpp:1009]: extract a function to check

Comment (Lloyd-Pottiger) [reply dbms/src/Interpreters/Context.cpp:1009]: good idea! thanks~

**Review by ywqzzy — APPROVED** [review]
LGTM

**Review by SeaRise — APPROVED** [review]
LGTM

File: dbms/src/Interpreters/Context.cpp [hunk 2]
```diff
@@ -1733,6 +1756,7 @@ void Context::setDefaultProfiles(const Poco::Util::AbstractConfiguration & confi
     shared->default_profile_name = config.getString("default_profile", "default");
     shared->system_profile_name = config.getString("system_profile", shared->default_profile_name);
     setSetting("profile", shared->system_profile_name);
+    is_config_loaded = true;
```
Comment (ywqzzy) [inline dbms/src/Interpreters/Context.cpp:1759]: Add a log to show the config is loaded

Comment (Lloyd-Pottiger) [reply dbms/src/Interpreters/Context.cpp:1761]: added in server.cpp

File: dbms/src/Interpreters/Context.cpp [hunk 3]
```diff
@@ -1004,8 +1004,21 @@ ASTPtr Context::getCreateDatabaseQuery(const String & database_name) const
     return shared->databases[db]->getCreateDatabaseQuery(*this);
 }
 
+void Context::checkIsConfigLoaded() const
+{
+    if (shared->application_type == ApplicationType::SERVER && !is_config_loaded)
+    {
+#ifndef NDEBUG
```
[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply dbms/src/Interpreters/Context.cpp:1011]: I have test serval times, there are no crash during start in debug mode. So maybe it is safe to always throw a exception, I will fix it.

File: dbms/src/Interpreters/Context.cpp [hunk 4]
```diff
@@ -1004,8 +1004,21 @@ ASTPtr Context::getCreateDatabaseQuery(const String & database_name) const
...
+        LOG_ERROR(shared->log, "Configuration are used before load from configure file tiflash.toml, so the user config may not take effect. There are must be some bugs. Please open an issue in https://github.com/pingcap/tiflash.");
+#else
+        throw Exception("Configuration are used before load from configure file tiflash.toml, so the user config may not take effect.", ErrorCodes::LOGICAL_ERROR);
+#endif
```
Comment (Lloyd-Pottiger) [inline dbms/src/Interpreters/Context.cpp:1015]:
```suggestion
        throw Exception("Configuration are used before load from configure file tiflash.toml, so the user config may not take effect.", ErrorCodes::LOGICAL_ERROR);
```

File: dbms/src/Interpreters/Context.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply dbms/src/Interpreters/Context.cpp]: fixed.

File: dbms/src/TestUtils/TiFlashTestEnv.cpp [hunk 5]
```diff
@@ -83,6 +83,9 @@ void TiFlashTestEnv::addGlobalContext(Strings testdata_path, PageStorageRunMode
     KeyManagerPtr key_manager = std::make_shared<MockKeyManager>(false);
     global_context->initializeFileProvider(key_manager, false);
 
+    // do not need to load config file in unit test, just set `is_config_loaded = true` to skip checkIsConfigLoaded
+    global_context->is_config_loaded = true;
```
Comment (SeaRise) [inline dbms/src/TestUtils/TiFlashTestEnv.cpp:87]: this line will build fail in release
```
dbms/src/TestUtils/TiFlashTestEnv.cpp:87:21: error: 'is_config_loaded' is a private member of 'DB::Context'
global_context->is_config_loaded = true
```

**Pull request merged**